MekHQ/src/mekhq/gui/adapter/TOEMouseAdapter.java

```diff
@@ -242,20 +351,35 @@ public void actionPerformed(ActionEvent action) {
                     MekHQ.triggerEvent(new OrganizationChangedEvent(singleForce));
                 }
             }
-        } else if (command.contains("REMOVE_FORCE")) {
+        } else if (command.contains(TOEMouseAdapter.REMOVE_FORCE)) {
             for (Force force : forces) {
                 if (null != force && null != force.getParentForce()) {
-                    if (0 != JOptionPane.showConfirmDialog(
+                    if (JOptionPane.YES_OPTION != JOptionPane.showConfirmDialog(
                             null,
                             "Are you sure you want to delete "
                                     + force.getFullName() + "?",
                                     "Delete Force?", JOptionPane.YES_NO_OPTION)) {
                         return;
                     }
+                    for (UUID id : force.getAllUnits()) {
+                        Unit unit = gui.getCampaign().getUnit(id);
+                        if (unit != null) {
+                            for (UUID oldShipId : unit.getTransportShipId().keySet()) {
```

Comment:
This and the handler for REMOVE_UNIT seem to have identical logic for individual units. Let's unify that into a single worker function.

Reply:
Fixed.